### PR TITLE
Feat(eos_designs):  Replace wan_role checks with helper cached_property utils

### PR DESCRIPTION
**Pull request by ayushmittal-arista**
## Change Summary
Introduced 3 properties: is_wan_router, is_wan_server, is_wan_client

Replaced:
  wan_role is not None -> is_wan_router
  wan_role is None -> not is_wan_router
  
  wan_role == server -> is_wan_server
  wan_role != server -> not is_wan_server
  and so on.

## Related Issue(s)

Fixes #

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Replace wan_role based checks with more defined functions

## How to test
no changes in output after running molecule 

### User Checklist

- N/A

### Repository Checklist

- [ ] My code has been rebased from devel before I start
- [ ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 1]
```diff
@@ -41,6 +41,27 @@ def wan_role(self: SharedUtils) -> str | None:
             raise AristaAvdError("'wan_role' client requires 'evpn_role' client.")
         return wan_role
 
+    @cached_property
+    def is_wan_router(self) -> bool:
+        if self.wan_role:
+            return True
+        else:
+            return False
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py]:
```suggestion
        return bool(self.wan_role)
```

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 2]
```diff
@@ -41,6 +41,27 @@ def wan_role(self: SharedUtils) -> str | None:
...
+
+    @cached_property
+    def is_wan_server(self) -> bool:
+        if self.wan_role == "server":
+            return True
+        else:
+            return False
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py]:
```suggestion
        return self.wan_role == "server":
```

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 3]
```diff
@@ -41,6 +41,27 @@ def wan_role(self: SharedUtils) -> str | None:
...
+
+    @cached_property
+    def is_wan_client(self) -> bool:
+        if self.wan_role == "client":
+            return True
+        else:
+            return False
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py]:
```suggestion
        return self.wan_role == "client":
```

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py [hunk 4]
```diff
@@ -41,6 +41,18 @@ def wan_role(self: SharedUtils) -> str | None:
             raise AristaAvdError("'wan_role' client requires 'evpn_role' client.")
         return wan_role
 
+    @cached_property
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py:44]: so not related to your PR but line 31 above:
```
if self.underlay_router is False or self.wan_mode is None:
```

`self.wan_mode` has a default value and will never be None

could you take advantage of this PR to replace it with

```
if self.underlay_router is False:
```

**Review by gmuloc — APPROVED** [review]
LGTM - would be cool if you can address the comment I left to clean up the useless code (not from your PR but from another time ;) )

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py:44]: done

**Pull request merged**